src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -8602,6 +8810,1964 @@
+            }
+
+            return ""Blue"";
+        }
+    }
+}
+
+class Program
+{
+    static void Main()
+    {
+        var c1 = new Color.C1(default);
+        object val = c1.M1(new Color.Red());
+        System.Console.Write(val);
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: @"Red").VerifyDiagnostics(
+                // (6,28): warning CS8907: Parameter 'Color' is unread. Did you forget to use it to initialize the property with that name?
+                //     public class C1 (Color Color)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "Color").WithArguments("Color").WithLocation(6, 28)
+                );
+
+            Assert.Empty(comp.GetTypeByMetadataName("Color+C1").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void ParameterCapturing_020_ColorColor_MemberAccess_Instance_FieldPropertyEvent(
+            [CombinatorialValues(
+                @"string Red { get; } = ""Red"";",
+                @"event System.Func<string> Red = () => ""Red"";",
+                @"string Red = ""Red"";"
+                )]
+            string member)
+        {
+            var source = @"
+class Color
+{
+    public " + member + @"
+
+    public class C1 (Color Color)
+    {
+        public object M1() {var val = Color.Red; return val; }
+    }
+}
+
+class Program
+{
+    static void Main()
+    {
+        var c1 = new Color.C1(new Color());
+        object val = c1.M1();
+        System.Console.Write(val is System.Func<string> d ? d() : val);
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: @"Red").VerifyDiagnostics();
+
+            Assert.NotEmpty(comp.GetTypeByMetadataName("Color+C1").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+        }
+
+        [Fact]
+        public void ParameterCapturing_021_ColorColor_MemberAccess_Instance_Method()
+        {
+            var source = @"
+class Color
+{
+    public string Red() => ""Red"";
+
+    public class C1 (Color Color)
+    {
+        public object M1() => Color.Red();
+    }
+}
+
+class Program
+{
+    static void Main()
+    {
+        var c1 = new Color.C1(new Color());
+        object val = c1.M1();
+        System.Console.Write(val);
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: @"Red").VerifyDiagnostics();
+
+            Assert.NotEmpty(comp.GetTypeByMetadataName("Color+C1").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+        }
+
+        [Fact]
+        public void ParameterCapturing_022_ColorColor_MemberAccess_Instance_Method()
```

Comment:
>ParameterCapturing_022_ColorColor_MemberAccess_Instance_Method [](http://example.com/codeflow?start=20&length=62)

Consider also testing when both extension method and instance method are available.